particles/Tutorial/Javascript/Demo/source/TTTGame.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+/* global defineParticle */
+defineParticle(({SimpleParticle, html}) => {
+
+  const winningSequences = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]
+  ];
+
+  const template = html`
+It is your turn <span>{{playerDetails}}</span>.
+<div slotid="gameSlot"></div>
+<div hidden="{{hideCongrats}}">Congratulations <span>{{winnerName}}</span>!</div>`;
+
+  return class extends SimpleParticle {
+
+    get template() {
+      return template;
+    }
+
+    shouldRender({gameState, playerOne, playerTwo}) {
+      return gameState && playerOne.id === 0 && playerTwo.id === 1;
+    }
+
+    update({gameState, playerOne, playerTwo, playerOneMove, playerTwoMove, events}) {
+
+      if (playerOne.id != 0) {
+        this.set('playerOne', {name: playerOne.name, avatar: playerOne.avatar, id: 0});
+      }
+
+      if (!playerTwo.id) {
+        this.set('playerTwo', {name: playerTwo.name, avatar: playerTwo.avatar, id: 1});
+      }
+
+      if (!gameState) {
+        this.newGame();
+      }
+
+      if (gameState && gameState.gameOver && events ) {
+        const reset = events.find(e => e.type === 'reset');
```

Comment:
I suggest factoring this code and the next conditional-body into separate functions. 

It's useful if it's easy to see from `update` the various states that can exist (like `gameState && gameState.gameOver && events`) and have simple named tasks triggered by those states.

`update` is intended to be like a switch board for reactions.